lib/Connection.js

```diff
@@ -107,7 +107,8 @@ class Connection extends EventEmitter {
     }
   }
 
-  _onClose() {
+  _onClose(code, reason) {
+    if (code) this.emit('ws.closed', [code, reason]);
```

Comment:
1. let's emit as the very last line in the method. This way, if there's an exception in user-land code, we won't be left in a non-consistent state.
2. We should emit unconditionally
3. Let's not emit any details for now

Reply:
done, hopefully i am not misunderstand again.